### ISSUE TITLE
Greg wants to do a pedalboard, how will this work?

### ISSUE THREAD

**Issue by elanhickler** [RobinSchmidt/RS-MET#97]
We need DSP for these:

Compressor
Gate
Chorus
Flanger
Phaser
Delay
Reverb (could always be handled with convolution, since I’ve made so many reverb impulses by now)
Convolution (this could be used for spring reverbs, cabinet emulation, and other reverb effect modules)

1. Will greg be paying your per-product cost for the DSP if we use your library for those things?
2. Do you have complete classes/code for these things?

I will do all the programming, unless we need to hire you for some custom DSP.

**Comment by elanhickler** [comment]
OrangeTreeSamples: https://www.orangetreesamples.com/

Not sure about the budget, we need a custom UI for a pedalboard where you can have virtual cables and connect anything to anything. We were planning on paying Jim for that, using of JUCE's VST host as a starting point.

**Comment by elanhickler** [comment]
wow this is a nice selection

![image](https://user-images.githubusercontent.com/9116575/31471152-2bee43c8-ae9c-11e7-8c88-bfa976b4bf95.png)

**Comment by RobinSchmidt** [comment]
ahhh - i wanted to check it out more before i point you to that. there are a lot of loose ends and incomplete implementations. but yes, that's most of collection of standard effects. maybe we should see it more as starting point...

**Comment by RobinSchmidt** [comment]
this "quadrifex" thing is one the plugins that i didn't get to publish, so far

**Comment by elanhickler** [comment]
yes, each effect will be a starting point, we will not be using any of them "as is" most likely.

**Comment by RobinSchmidt** [comment]
yeah - i'm especially interested in doing some more research-and-development for reverb (widely considered as supreme discipline in audio dsp). i have once published some ideas in a paper, but did not yet really make any good use of them:
http://www.rs-met.com/documents/dsp/GeneralizedHadamardTransform.pdf
